routes/clucks.js

```diff
@@ -0,0 +1,57 @@
+const path = require('path');
+const multer = require('multer');
+const express = require('express');
+const router = express.Router();
+const knex = require('../db');
+
+const UPLOADS_DIR = 'uploads';
+const upload = multer({dest: path.join(__dirname, '..', 'public', UPLOADS_DIR)});
+
+// PATH: /clucks/new VERB: GET Serves form for creating clucks
+router.get('/new', (request, response) => {
+  response.render('clucks/new');
+});
+
+// localhost:7000/clucks/:id
+router.get('/:id', (request, response) => {
+  const id = request.params.id;
+  knex
+    .first()
+    .from('clucks')
+    .where({id})  //{id} === {id: id}
+    .then(cluck => {
+      response.render('clucks/show',{cluck});
+    })
+    .catch(error => response.send(error));
+});
+
+// PATH: /clucks VERB: POST Creating new clucks
+router.post('/', upload.single('picture'), (request, response) => {
+  const username = request.body.username;
+  const content = request.body.content;
+  // const image_url = request.body.image_url;
+
+  const filename = request.file.filename;
```

Comment:
This code only works if the user submits the form with a file to upload.

If there is no file, then `multer` does not add a `file` object to the `request` object.
This means that `request.file` is undefined. And `undefined.filename` leads to a error and your code crashing.
Here, I would just write this code slightly differently:

```javascript
let image_url = '';
if (request.file) {
   const filename = request.file.filename;
   image_url = path.join(UPLOADS_DIR, filename);
}
```
And then keep the rest of your code the same. This will prevent this error from happening